app/code/Magento/Quote/etc/fieldset.xml

```diff
@@ -205,6 +205,9 @@
             <field name="qty">
                 <aspect name="to_order_item" targetField="qty_ordered" />
             </field>
+            <field name="quote_item_id">
+                <aspect name="to_order_item_id" targetField="quote_item_id" />
```

Comment:
Considering that according to this pull request, quote item and quote address item should be processed in a different way (address item requires additional conversions), what do you think about adding a  `quote_convert_address_item` fieldset where`quote_item_id` is populated from the `quote_item_id` field instead of `id`:

         <fieldset id="quote_convert_address_item">
             <field name="quote_item_id">
                 <aspect name="to_order_item" />
             </field>
         </fieldset>

Then you can add additional data to $orderItemData in ToOrderItem.php after the initial data retrival:

        if ($item instanceof \Magento\Quote\Model\Quote\Address\Item) {
             $orderItemData= array_merge(
                 $orderItemData,
                $this->objectCopyService->getDataFromFieldset(
                    'quote_convert_address_item',
                    'to_order_item',
                    $item
                )
            );
        }

This approach will enable easy implementation for further conversion differences of address item, simply by altering the introduced `quote_convert_address_item` fieldset

Reply:
@sivaschenko i will add this implementation, Thanks for the suggestion.

Reply:
@maheshWebkul721 👍 thank you